Add vendor_resp.normalized_applicant to mock

pkarman opened the pull request:
**Why**: Pretend that the vendor can return "corrected"
values for applicant attributes.

zachmargolis commented on lib/proofer/vendor/mock.rb:
```diff
@@ -132,35 +132,66 @@ def fail_confirmation_with_bad_phone(session_id)
+      # rubocop:enable Metrics/AbcSize
+
+      def normalize(str)
+        return str.upcase unless str.nil?
```
looks like reek complained about this?

what about using a ternary?

```
str ? str.upcase : ''
```

pkarman replied:
I do like the ternary but had been trying to ween myself off them. ;)

In this case it does seem simpler. changed now.

zachmargolis reviewed (approved):
LGTM

Commit:
Add vendor_resp.normalized_applicant to mock

**Why**: Pretend that the vendor can return "corrected"
values for applicant attributes.
